### PR TITLE
Support detecting terminal profile with potentially unsafe paths and add an opt-in

### LINKED ISSUE #167721: Bring back unsafe terminal profile paths with a confirmation step
The idea is to bring back cygwin and add other shells like cmder/msys2 and any other profiles that may be in "unsafe" paths (ie. not those owned by the user or admin) and present them in the list:

image

Note that they're in the `detected` section, not the `profiles` section, so it won't show up unless it's created via the select default profile button. It's not clear to me whether we need a warning icon or warning text here or not, but clicking it would get a notification warning before proceeding which is the main protection.

Code for the screenshot:

```ts
		detectedProfiles.set('Cygwin $(warning)', {
			path: [
				'C:\\cygwin64\\bin\\bash.exe',
				'C:\\cygwin\\bin\\bash.exe'
			],
			args: ['--login'],
			isAutoDetected: true
		});
```

### PR DESCRIPTION
**Pull request by Tyriar**
Fixes #167721

Some background on this, there was a CVE recently about picking up shells with paths that could be written to by other users on Windows in a shared user environment. The cygwin profile was removed and the git bash profile lost a single path it used to pick up:

- `C:\Cygwin64\bin\bash.exe`
- `C:\Cygwin\bin\bash.exe`
- `C:\ProgramData\scoop\apps\git-with-openssh\current\bin\bash.exe`

The PR for reference is https://github.com/microsoft/vscode/issues/160827.

---

This change adds the ability to pick unsafe paths up as a "detected profile", similar to Windows PowerShell when PowerShell is not installed, it will not show up in new terminal with profile list by default:

image

But we do make it more convenient to set up:
image
image

From there the user can select the item to make it the default or click the gear to just add it to settings.

This is what that looks like now if you have Cygwin or MSYS2 (new) installed:

image
image

Notice the warning icon next to the exe as an extra indicator that this one is special. When you click on either the profile or the gear, it will present this notification:

image

Notice that Cancel is the default which is atypical, not sure whether we should swap it?

When accepting:

image

This change will also let us support cmder https://github.com/microsoft/vscode/issues/125387, but that needs some more functionality added to profiles and I didn't want to make this PR more complex than it needed to be.

### CODE REVIEW

**Comment by sbatten** [comment]
Can the warning icons in the quickpick show a hover to explain the warning without selecting the item?

Should the settings editor also have a warning or is this only to protect you if in the easy route?

For cancel, I see why having it the default makes sense, but we don't do that for workspace trust (don't trust in that case) so I'm not sure if it align with previous decisions.

**Comment by Tyriar** [comment]
> Can the warning icons in the quickpick show a hover to explain the warning without selecting the item?

I don't think that's possible currently

> Should the settings editor also have a warning or is this only to protect you if in the easy route?

This is just for the easy route where we detect it, if the user added the path manually they wanted to do it and it's their own fault if it's dangerous (probably not).

> For cancel, I see why having it the default makes sense, but we don't do that for workspace trust (don't trust in that case) so I'm not sure if it align with previous decisions.

👍 will switch

File: src/vs/workbench/contrib/terminal/browser/terminalProfileQuickpick.ts [hunk 1]
```diff
@@ -197,20 +203,45 @@ export class TerminalProfileQuickpick {
...
+	private async _isProfileSafe(profile: ITerminalProfile | IExtensionTerminalProfile): Promise<boolean> {
+		if (!('isUnsafePath' in profile)) {
+			return true;
+		}
+
+		// Get the user's explicit permission to use a potentially unsafe path
+		return await new Promise<boolean>(r => {
+			const handle = this._notificationService.prompt(
+				Severity.Warning,
+				nls.localize('unsafePathWarning', 'This profile uses the potentially unsafe path {0} that another user can write to. Are you sure you want to use it?', `"${profile.path}"`),
```
Comment (sbatten) [inline src/vs/workbench/contrib/terminal/browser/terminalProfileQuickpick.ts:224]: Motivations for suggesting a slight change in phrasing:
- The image shows a weird line break in the middle of a sentence.
- Sentence ending in "to" feels awkward.

`This profile uses a potentially unsafe path that can be modified by another user: {0}. Are you use you want to use it?`

**Review by sbatten — DISMISSED** [review]

**Comment by sbatten** [comment]
For the warning icon, it doesn't seem too valuable without any hover text. It seems like its trying to shortcut the information to the user, but since we can't do that, I'm not sure it is necessary at all. The notification seems sufficient.

**Review by sbatten — APPROVED** [review]

**Pull request merged**